pkg/codegen/hcl2/functions.go

```diff
@@ -196,6 +196,21 @@ var pulumiBuiltins = map[string]*model.Function{
 		}},
 		ReturnType: model.StringType,
 	}),
+	"secret": model.NewFunction(model.GenericFunctionSignature(
+		func(args []model.Expression) (model.StaticFunctionSignature, hcl.Diagnostics) {
+			valueType := model.Type(model.DynamicType)
+			if len(args) == 1 {
+				valueType = args[0].Type()
+			}
+
+			return model.StaticFunctionSignature{
+				Parameters: []model.Parameter{{
+					Name: "value",
+					Type: valueType,
+				}},
+				ReturnType: model.NewOutputType(valueType),
+			}, nil
+		})),
```

Comment:
I added this function, but apparently I did not add support for it to the code generators. Should be straightforward.